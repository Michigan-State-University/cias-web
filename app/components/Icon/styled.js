import React from 'react';
import ReactSVG from 'react-inlinesvg';
import styled from 'styled-components';

import { margin } from '../BaseComponentStyles';

export const SVG = styled(({ fill, stroke, ...props }) => (
  <ReactSVG {...props} />
))`
  *[fill^='#'] {
    fill: ${({ fill }) => fill};
  }
  *[stroke^='#'] {
    stroke: ${({ stroke }) => stroke};
  }

  cursor: ${({ onClick }) => (onClick ? 'pointer' : 'initial')};
  ${margin};
`;
