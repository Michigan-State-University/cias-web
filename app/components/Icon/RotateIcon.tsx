import { FunctionComponent } from 'react';
// @ts-ignore
import styled from 'styled-components';

import Icon from './index';

export type Props = {
  $rotate: boolean;
};

// @ts-ignore
export const RotateIcon: FunctionComponent<Props> = styled(Icon)`
  transition: transform 0.4s ease;
  ${({ $rotate }: Props) => ($rotate ? 'transform: rotate(180deg);' : '')}
`;
