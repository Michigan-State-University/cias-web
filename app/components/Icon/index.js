/**
 *
 * Icon
 *
 */

import React from 'react';
import * as PropTypes from 'prop-types';

import * as Styled from './styled';

const Icon = ({ src, className, fill, stroke, alt, onClick, ...restProps }) => (
  <Styled.SVG
    src={src}
    className={className}
    fill={fill}
    stroke={stroke}
    alt={alt}
    onClick={onClick}
    {...restProps}
  />
);

Icon.propTypes = {
  className: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string,
  onClick: PropTypes.func,
};

Icon.defaultProps = {
  alt: '',
};

export default Icon;
