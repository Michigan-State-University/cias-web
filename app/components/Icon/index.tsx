/**
 *
 * Icon
 *
 */

import React, { CSSProperties } from 'react';
import { Props as SVGProperies } from 'react-inlinesvg';

import {
  MarginProps,
  LayoutProps,
  TextProps,
} from 'components/BaseComponentStyles';

import * as Styled from './styled';

export type Props = {
  src: SVGElement | string;
  alt?: string;
  inline?: boolean;
  wrapperProps?: LayoutProps & TextProps;
} & Omit<SVGProperies, 'src'> &
  CSSProperties &
  MarginProps &
  LayoutProps;

const Icon = ({
  src,
  alt = '',
  inline = false,
  wrapperProps,
  ...restProps
}: Props) => (
  <Styled.SvgWrapper inline={inline} {...wrapperProps}>
    <Styled.SVG src={src} alt={alt} {...restProps} />
  </Styled.SvgWrapper>
);

export default Icon;
