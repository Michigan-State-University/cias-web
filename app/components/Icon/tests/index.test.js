/**
 *
 * Tests for Icon
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import Icon from '../index';

describe('<Icon />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Icon src="mock.svg" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<Icon />);
    expect(container).toMatchSnapshot();
  });
});
