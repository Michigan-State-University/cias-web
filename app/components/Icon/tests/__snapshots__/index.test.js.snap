// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Should render and match the snapshot 1`] = `
<div>
  <img />
</div>
`;
